src/libxtp/gwbse/rpa.cc

```diff
@@ -81,5 +85,92 @@ Eigen::MatrixXd RPA::calculate_epsilon(double frequency) const {
 template Eigen::MatrixXd RPA::calculate_epsilon<true>(double frequency) const;
 template Eigen::MatrixXd RPA::calculate_epsilon<false>(double frequency) const;
 
+RPA::rpa_eigensolution RPA::Diagonalize_H2p() const {
+  const Index lumo = _homo + 1;
+  const Index n_occ = lumo - _rpamin;
+  const Index n_unocc = _rpamax - lumo + 1;
+  const Index rpasize = n_occ * n_unocc;
+
+  Eigen::VectorXd AmB = Calculate_H2p_AmB();
+  Eigen::MatrixXd ApB = Calculate_H2p_ApB();
+
+  Eigen::MatrixXd C = AmB.cwiseSqrt().asDiagonal() * ApB * AmB.cwiseSqrt().asDiagonal();
+  tools::EigenSystem result = Diagonalize_H2p_C(C);
+  
+  // Omega has to have correct size otherwise MKL does not rescale for Sqrt
+  Eigen::VectorXd omega;
+  omega = Eigen::VectorXd::Zero(rpasize);
+  omega = result.eigenvalues().cwiseSqrt();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'omega' is reassigned a value before the old one has been used.](https://app.codacy.com/app/votca-package/xtp/pullRequest?prid=4286871)